Source/Core/DolphinWX/InputConfigDiag.h

```diff
@@ -83,6 +84,38 @@ class PadSettingCheckBox : public PadSetting
 	ControllerEmu::ControlGroup::Setting* const setting;
 };
 
+class InputEventFilter : public wxEventFilter
+{
+public:
+	InputEventFilter()
+		: m_block(false)
+	{
+		wxEvtHandler::AddFilter(this);
+	};
+
+	~InputEventFilter()
+	{
+		wxEvtHandler::RemoveFilter(this);
+	};
+
+	int FilterEvent(wxEvent& event) override;
+
+	void BlockEvents(bool block) { m_block = block; };
+
+private:
+	static inline bool ShouldCatchEventType(wxEventType type)
+	{
+		// Can't use a switch... "expression must have a constant value"
+		return type == wxEVT_KEY_DOWN || type == wxEVT_KEY_UP ||
+			type == wxEVT_CHAR || type == wxEVT_CHAR_HOOK ||
+			type == wxEVT_LEFT_DOWN || type == wxEVT_LEFT_UP ||
+			type == wxEVT_MIDDLE_DOWN || type == wxEVT_MIDDLE_UP ||
+			type == wxEVT_RIGHT_DOWN || type == wxEVT_RIGHT_UP;
+	}
+
+	bool m_block;
```

Comment:
`bool m_block = false`

Then the initializer list in the constructor can be removed.